packages/fxa-settings/src/components/TotpInputGroup/index.tsx

```diff
@@ -0,0 +1,234 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import classNames from 'classnames';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { CodeArray } from '../FormVerifyTotp';
+
+export type TotpInputGroupProps = {
+  codeArray: CodeArray;
+  codeLength: 6 | 8;
+  inputRefs: React.MutableRefObject<React.RefObject<HTMLInputElement>[]>;
+  localizedInputGroupLabel: string;
+  setCodeArray: React.Dispatch<React.SetStateAction<CodeArray>>;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+  errorMessage?: string;
+};
+
+type SingleInputProps = {
+  index: number;
+};
+
+export const TotpInputGroup = ({
+  codeArray,
+  codeLength,
+  inputRefs,
+  localizedInputGroupLabel,
+  setCodeArray,
+  setErrorMessage,
+  errorMessage = '',
+}: TotpInputGroupProps) => {
+  const focusOnNextInput = (index: number) => {
+    inputRefs.current[index + 1].current?.focus();
+  };
+
+  const focusOnPreviousInput = (index: number) => {
+    inputRefs.current[index - 1].current?.focus();
+  };
+
+  const focusOnSpecifiedInput = (index: number) => {
+    inputRefs.current[index].current?.focus();
+  };
+
+  const handleBackspace = async (index: number) => {
+    let currentCodeArray = [...codeArray];
+    currentCodeArray[index] = undefined;
+    await setCodeArray(currentCodeArray);
```

Comment:
Couple nits:
* Can `currentCodeArray` become a `const` (also in delete fn)?
* Does `setCodeArray` need an `await` (also in delete fn)?
* Drop `focusOnSpecifiedInput(index);` to its own line after the `else`?

Reply:
Yes to 1 and 3, but `await` is needed for `setCodeArray` else there are glitches with the focus change in the next step.

Reply:
🤔 We might need a comment above that then. When I was in this branch I saw my linter saying "await" wasn't needed. Not sure how that works since it is sync.